test/geopackage/SQLite_Test.cpp

```diff
@@ -0,0 +1,111 @@
+#include <gtest/gtest.h>
+
+#include "NGen_SQLite.hpp"
+#include "FileChecker.h"
+
+using namespace geopackage;
+
+class SQLite_Test : public ::testing::Test
+{
+  protected:
+    void SetUp() override
+    {
+        this->path = utils::FileChecker::find_first_readable({
+            "test/data/routing/gauge_01073000.gpkg",
+            "../test/data/routing/gauge_01073000.gpkg",
+            "../../test/data/routing/gauge_01073000.gpkg"
+        });
+
+        if (this->path.empty()) {
+            FAIL() << "can't find gauge_01073000.gpkg";
+        }
+    }
+
+    void TearDown() override {};
+    
+    std::string path;
+    
+};
+
+TEST_F(SQLite_Test, sqlite_access_test)
+{
+    sqlite db {this->path};
+    // user wants metadata
+    EXPECT_TRUE(db.has_table("gpkg_contents"));
+    EXPECT_FALSE(db.has_table("some_fake_table"));
+}
+
+TEST_F(SQLite_Test, sqlite_query_test)
+{
+    sqlite db {this->path};
+
+    if (db.connection() == nullptr) {
+        FAIL() << "database is not loaded";
+    }
+
+    // user provides a query
+    const std::string query = "SELECT * FROM gpkg_contents LIMIT 1";
+    sqlite_iter iter = db.query(query);
+
+    EXPECT_EQ(iter.num_columns(), 10);
+    EXPECT_EQ(iter.columns(), std::vector<std::string>({
+        "table_name",
+        "data_type",
+        "identifier",
+        "description",
+        "last_change",
+        "min_x",
+        "min_y",
+        "max_x",
+        "max_y",
+        "srs_id"
```

Comment:
Does SQLite's API guarantee that columns are returned in a specified order?

Reply:
I'm actually not sure if the C library guarantees column ordering -- that's a good question. Intuitively, I think columns are indexed based off the meta table describing the table schema, and these names are accessed using `sqlite_column_name()` with the column index as an argument, so possibly?